### PR TITLE
Disallow querying delta lake properties table

### PR DESCRIPTION
**Pull request by findinpath**
Extracted relevant commits for disallowing the querying of the `$properties` table on the delta lake tables in hive from the PR https://github.com/trinodb/trino/pull/10441

### CODE REVIEW

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java [hunk 1]
```diff
@@ -60,6 +63,9 @@
         if (table.isEmpty()) {
             throw new TableNotFoundException(tableName);
         }
+        if (isDeltaLakeTable(table.get())) {
+            throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, "Cannot query Delta Lake table");
```
Comment (phd3) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:67]: nit: would be good to mention table name in error message

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java [hunk 2]
```diff
@@ -425,12 +425,14 @@ public HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName
...
-        if (table.isEmpty()) {
-            return null;
-        }
+        return metastore.getTable(new HiveIdentity(session), tableName.getSchemaName(), tableName.getTableName())
+                .map(table -> getTableHandle(session, tableName, table))
+                .orElse(null);
+    }
 
-        if (isDeltaLakeTable(table.get())) {
+    private HiveTableHandle getTableHandle(ConnectorSession session, SchemaTableName tableName, Table table)
```
Comment (phd3) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:433]: I'm not sure if having a separate method is super useful here, we could store `Optional<Table>` as name optionalTable OR just do `metastore.getTable().orElse(null)`. but don't feel very strongly.

**Review by phd3 — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java (hunk 2 above)
Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java:433]: You are right. The redirection in the code to another method is not necessary here.
I modified the code to have the logic of the `getTableHandle(...)` method contained where it was initially.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java [hunk 3]
```diff
@@ -60,6 +64,9 @@
         if (table.isEmpty()) {
             throw new TableNotFoundException(tableName);
         }
+        if (isDeltaLakeTable(table.get())) {
+            throw new TrinoException(HIVE_UNSUPPORTED_FORMAT, format("Cannot query Delta Lake table '%s'", sourceTableName));
```
Comment (findepi) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:68]: Same for `$partitions`

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:67]: I modified the exception message to its original format to be in sync with the other places where the exception is being thrown.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java (hunk 3 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java]: This is marked resolved. what is the resolution? I am not seeing changes in `PartitionsSystemTableProvider`, but i guess i am missing something.

Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java]: I was planning to implement the functionality in the `PartitionsSystemTableProvider` , but dropped the idea after seeing the call to `io.trino.plugin.hive.HiveMetadata#getTableHandle` method, which has  already the check for delta lake format.

See 
https://github.com/trinodb/trino/blob/master/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L433-L435

and do note that the test has been updated to check the query of `$partitions` system table as well

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java (hunk 1 above)
Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:67]: I agree with @phd3  we should provide table name in the message.

**Review by findepi — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java (hunk 1 above)
Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:67]: Good point. I'll make a separate commit then to update everywhere the exception message.

Comment (findepi) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/PropertiesSystemTableProvider.java:67]: agreed, the `io.trino.plugin.hive.HiveMetadata#getTableHandle` can (and should) be updated in a separate commit.

**Comment by phd3** [comment]
Merged, thanks!

**Pull request merged**